m4/ax_append_flag.m4

```diff
@@ -0,0 +1,71 @@
+# ===========================================================================
+#      https://www.gnu.org/software/autoconf-archive/ax_append_flag.html
+# ===========================================================================
+#
+# SYNOPSIS
+#
+#   AX_APPEND_FLAG(FLAG, [FLAGS-VARIABLE])
+#
+# DESCRIPTION
+#
+#   FLAG is appended to the FLAGS-VARIABLE shell variable, with a space
+#   added in between.
+#
+#   If FLAGS-VARIABLE is not specified, the current language's flags (e.g.
+#   CFLAGS) is used.  FLAGS-VARIABLE is not changed if it already contains
+#   FLAG.  If FLAGS-VARIABLE is unset in the shell, it is set to exactly
+#   FLAG.
+#
+#   NOTE: Implementation based on AX_CFLAGS_GCC_OPTION.
+#
+# LICENSE
+#
+#   Copyright (c) 2008 Guido U. Draheim <guidod@gmx.de>
+#   Copyright (c) 2011 Maarten Bosmans <mkbosmans@gmail.com>
+#
+#   This program is free software: you can redistribute it and/or modify it
+#   under the terms of the GNU General Public License as published by the
+#   Free Software Foundation, either version 3 of the License, or (at your
+#   option) any later version.
+#
+#   This program is distributed in the hope that it will be useful, but
+#   WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+#   Public License for more details.
+#
+#   You should have received a copy of the GNU General Public License along
+#   with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+#   As a special exception, the respective Autoconf Macro's copyright owner
```

Comment:
I am not a lawyer... is this exception enough for us to include GPLd code in an otherwise MIT licensed project?

Reply:
I am not a lawyer either, ugh, but, the m4 directory is full of alternate licenses.
In general there is a line between "build tools" and "their output".
Perhaps it is explicit, granted.
For example we use gcc but that does not affect our license.
Mono's license presumably does not affect its AOT output.

Reply:
The exception sounds adequate:
```
#   As a special exception, the respective Autoconf Macro's copyright owner
#   gives unlimited permission to copy, distribute and modify the configure
#   scripts that are the output of Autoconf when processing the Macro. You
#   need not follow the terms of the GNU General Public License when using
#   or distributing such scripts, even though portions of the text of the
#   Macro appear in them. The GNU General Public License (GPL) does govern
#   all other use of the material that constitutes the Autoconf Macro.
```

Reply:
Is this blocked on licensing?
@grendello already added `ax_check_compile_flag.m4`.

Reply:
This licensing gives you/us the right to use the output of those scripts freely, that's the most important aspect. If you were to modify this script, create another based on it *then* your creation (the new script) would have to be GPL. The configure script being the result of **running** automake which **uses** the m4 tool and the associated scripts, does not impose any licensing requirements on the product (the configure script) nor the user of that product (Mono). IANAL, of course, but this is the standard mode of use of automake+friends in software that uses autotools and ships with a multitude of licenses other than the GNU ones.